ignore.go

```diff
@@ -0,0 +1,119 @@
+package metafora
+
+import (
+	"container/heap"
+	"sync"
+	"time"
+)
+
+type ignoremgr struct {
+	incoming chan *timetask
+	mu       *sync.RWMutex
+	ignores  map[string]struct{}
+}
+
+func ignorer(tasks chan<- string, stop <-chan struct{}) *ignoremgr {
+	im := &ignoremgr{mu: &sync.RWMutex{}, ignores: make(map[string]struct{}), incoming: make(chan *timetask)}
+	go im.monitor(tasks, stop)
+	return im
+}
+
+func (im *ignoremgr) add(taskID string, until time.Time) {
+	// short circuit ignores that have already elapsed
+	if until.Before(time.Now()) {
+		return
+	}
+	im.mu.Lock()
+	im.ignores[taskID] = struct{}{}
+	im.incoming <- &timetask{time: until, task: taskID}
+	im.mu.Unlock()
+}
+
+func (im *ignoremgr) ignored(taskID string) (ignored bool) {
+	im.mu.RLock()
+	_, ok := im.ignores[taskID]
+	im.mu.RUnlock()
+
+	return ok
+}
+
+func (im *ignoremgr) monitor(tasks chan<- string, stop <-chan struct{}) {
+	times := timeheap{}
+	heap.Init(&times)
+	var next *timetask
+	for {
+		if times.Len() > 0 {
```

Comment:
It seems like this block could get simpler if you implemented Peek on your Heap?   But I maybe missing something..

https://github.com/lytics/toolbucket/blob/master/orderedtask/prioityqueue.go#L39